### ISSUE TITLE
rdflib no longer recognises an editable Document as editable

### ISSUE THREAD

**Issue by Vinnl** [linkeddata/rdflib.js#359]
I have a call to `UpdateManager.update()` on a file (`publicTypeIndex.ttl` on my Pod, with unchanged access controls), which was performed successfully with rdflib 0.20.3. However, with version 1.0.4, this suddenly resulted in the following error:

> Error: Exception in update: Error: Update: Loaded https://vincent.localhost:8443/settings/publicTypeIndex.ttlbut stil can't figure out what editing protcol it supports.

The primary difference between the two appears to be that the `editable` function in `update-manager.js` is called, which returns `'SPARQL'` in the working version, but `undefined` in the newer version.

Specifically, the working version hits [this code path](https://github.com/linkeddata/rdflib.js/commit/11519162df4d31067a5c175686a29532552f2bea#diff-0aaa1c3585187a3868b62f3bcdca96f4L103):

```javascript
    // Hack for the moment @@@@ 2016-02-12
    if (kb.holds(namedNode(uri), this.ns.rdf('type'), this.ns.ldp('Resource'))) {
      return 'SPARQL'
    }
```

Although Tim's comment says it was introduced by @dmitrizagidulin, [it was actually added by...](https://github.com/linkeddata/rdflib.js/commit/69d117dc698a6978c35964afb8dbe6e4a9956e4e#diff-971ef271c5851a3f095f55430133aa91R55) 

So the question is: what was the old code supposed to do, and what should be changed in the new code to make it work again?

**Comment by Vinnl** [comment]
Dug more deeply into this.

So there were three ways rdflib would detect whether it could send PUT requests:

1. Check if the target URL was known to be an `ldp:Resource`.
2. Check if the target URL has the header  `Accept-Patch: application/sparql-update`.
3. Check if the target URL has the header `MS-Author-Via: SPARQL`.

The first option has now been removed. The third option was the only way that _would_ work for Node Solid Server, which does not appear to send the header in option 2, but unfortunately, the Fetcher could not access that header due to CORS restrictions.

I've submitted a PR to Node Solid Server that allows apps access to the header from option 3: https://github.com/solid/node-solid-server/pull/1313

However, until that time, rdflib remains broken for cross-domain interaction with NSS.
